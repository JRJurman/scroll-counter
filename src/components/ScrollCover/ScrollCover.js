import { registerHtml } from 'tram-one'
import './ScrollCover.scss'
const html = registerHtml()

export default ({ color }) => {
  const coverStyle = `
    background: linear-gradient(to right, ${color}C9 30%, transparent 40%, transparent 60%, ${color}C9 70%);
  `
  return html`
    <div class="ScrollCover" style=${coverStyle} />
  `
}
