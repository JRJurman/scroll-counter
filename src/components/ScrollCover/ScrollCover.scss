.ScrollCover {
  width: 100%;
  height: 2em;
  margin-top: -2em;
  pointer-events: none;
  background: linear-gradient(to right, #f8ca9dc9 30%, transparent 40%, transparent 60%, #f8ca9dc9 70%);
  position: absolute;
}
