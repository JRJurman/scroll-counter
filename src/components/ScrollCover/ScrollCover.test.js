import ScrollCover from './ScrollCover'

describe('ScrollCover', () => {
  describe('with default props', () => {
    const wrapper = ScrollCover({ color: '#FFEEDD' })

    it('should match snapshot', () => {
      expect(wrapper.outerHTML).toMatchSnapshot()
    })
  })
})
