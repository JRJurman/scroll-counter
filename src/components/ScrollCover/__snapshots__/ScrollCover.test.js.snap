// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollCover with default props should match snapshot 1`] = `
"<div style=\\"
    background: linear-gradient(to right, #FFEEDDC9 30%, transparent 40%, transparent 60%, #FFEEDDC9 70%);
  \\" class=\\"ScrollCover\\"></div>"
`;
